### PR TITLE
Created Top Portion Of Landing Page

### LINKED ISSUE #73: Landing Page: Top Portion
### Dependency

- [x] #60 

### Overview
As a developer, we need to combine our components into our product so that we can deliver to stakeholders! For this issue, we will create the top portion of the landing page.

### Action Items

- [ ] Review the top portion of the landing page (this is everything above the CoP circles but does not include the CoP circles)
- [ ] Create the top portion
  - [ ] Note that, we preferably would not hard-code spaces (this is not responsive), but use the spacing utilities instead. Some classes have hard-coded spacing built in, so be sure to exchange them for the spacing utility classes.
- [ ] Release dependency on:
  - [ ] #79
  - [ ] #96 
  - [ ] #120

### Resources/Instructions
[Figma, Landing Page](https://www.figma.com/file/G5bOqhud6azbxyR9El9Ygp/Civic-Tech-Jobs?node-id=1%3A7)

### PR DESCRIPTION
**Pull request by NickTSA**
close #73 

- created top portion of landing page 
- added pages directory for lading page
- layered SVG images and created responsive design when scaling mobile to desktop screen sizes

NOTES: _design is not optimal for extra large screen size such as 2000px plus this can be updated to work a little better with current design but somethings ultimately will not scale proportionally. This is not a big concern at the moment since it is unusual to extended a window to this size._

### CODE REVIEW

**Comment by Aveline-art** [comment]
@NickTSA Please correct all status checks before requesting a review

**Comment by NickTSA** [comment]
@Aveline-art Ready for review

**Review by Aveline-art — CHANGES_REQUESTED** [review]
Good work on this! The page looks quite nice on desktop. I'll leave a higher review for this first pass, and go into specifics once the larger things are dealt with. A couple of ways to tighten up this code.

- ~~Using three separate sections to recreate the image is not sustainable, since if we decide to switch over to a different backdrop, we would need to redo the css code. Let's ask the design team to provide us with one whole image, and see if that is workable. That said, let's shelf the code you have now inside an unused file, maybe "landingpage-beta.js" and "_landingpage-beta.scss"~~ Never mind this, I will make a new issue for design to provide the image and we will investigate this as part of post-landing page clean up
- The second and first z-index would not be needed if the header is set with a positive z-index. Let's add a new file with z-index variables to our `Basics/` and import it into the header component. For an arbitrary high value, let's use 1030 and name it $z-header.
- Let's use the button component rather than the button tag, so that we can make edits on the component rather than the page level.
- The text needs to be responsive, as it shrinks when in mobile.

**Comment by NickTSA** [comment]
@Aveline-art I added the z-index into utilities. 

there is an issue with the button size not updating I tried the addClass prop and it doesn't work. I put the button-responsive class and it doesn't take effect for some reason. Other classes like "mt" work when using "addClass" not sure what the issue is here.

File: frontend/src/pages/landingPage/landingPage.scss [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+// }
+
+// code below isnt working working with Button component as it did with native button element
+
+.action-btn {
+  @include button-responsive("smtablet", "md");
+  @include button-responsive("lgtablet", "lg");
+  @include button-responsive("laptop", "lg");
+  @include button-responsive("desktop", "lg");
+}
```
Comment (Aveline-art) [inline frontend/src/pages/landingPage/landingPage.scss:70]: The screen sizes are not ordered from largest to smallest, so laptop lg will override the smaller size. This is what the current logic is:

If size is smtablet or smaller, use medium.
If size is lgtablet or smaller, use large.
If size is laptop or smaller, use large.
If size is desktop, use large.

Therefore, if the screen size is smtablet, it is smaller than laptop, so large is used. To get the behavior you want, you need to reorder the logic.

If size is desktop, use large.
If size is laptop or smaller, use large.
If size is lgtablet or smaller, use large.
If size is smtablet or smaller, use medium.

Since smtablet is smtablet or smaller, it will use medium.

The Design System Helper explains [this somewhat](https://aveline-art.github.io/CivicTechJobs-Documentation/developer/design-system/#responsive-mixins), although I might edit it a little so that it better applies to components with a default built in.

File: frontend/src/pages/landingPage/landingPage.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+          Together,
+          <br />
+          we can create greater civic change
+        </h1>
+        <p className="mb-5">
+          CivicTechJobs unites ambitious technology practitioners with volunteer
+          opportunities from a central hub of listings to build digital
+          products, programs, and services.{" "}
+        </p>
+        <Button addClass="action-btn">Join us</Button>
```
Comment (Aveline-art) [inline frontend/src/pages/landingPage/landingPage.js:32]: This does not set the default size to lg, so it uses the default of sm. Check out the [button file ](https://github.com/hackforla/CivicTechJobs/blob/main/frontend/src/components/Buttons/Buttons.js)to see the props that it takes and [here ](https://github.com/hackforla/CivicTechJobs/blob/588db284358ab4be178fae87ec0c6235f0134474/frontend/src/components/App.js#L102)for sample usage.

File: frontend/src/pages/landingPage/landingPage.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import { groupBackground } from "../../assets/images/images";
+import { IconDownArrow } from "../../assets/images/images";
+import { Button } from "../../components/Buttons/Buttons";
+
+import "./landingPage.scss";
```
Comment (Aveline-art) [inline frontend/src/pages/landingPage/landingPage.js:6]: This is causing an import bug. I sent an announcement over slack about it, but you might have missed it. I will paste it below so you can refer back to it whenever you need to.

> 
> @Sagun Shrestha @Narek Tsaturyan I think I know how some of the styles are being overridden. The main reason is that index.scss is compiled last, so any styles in that takes precedence over everything else. So instead of doing this for importing:
> 
> Footer.js
> ```
> import React, { Fragment } from "react";
> import { Button } from "../Buttons/Buttons";
> import "./_Footer.scss"; <----- DO NOT DO THIS!
> ```
> Do this instead:
> 
> index.scss'
> ```
> @use "components/normalize";
> @use "components/Basics" as *;
> @use "components/Buttons/Buttons" as *;
> @use "path/to/Footer.scss"; <------- DO THIS!
> ```
> This is my fault as I have been doing things the first way, but with your current and future issues please do it the second way. I will separately move all imports over to index.scss and fix any bugs that come up and merge the fix after your issues are done.

**Review by Aveline-art — COMMENTED** [review]
I added a couple of things to do to solve the bug. Let me know if you still need help!

**Comment by narek** [comment]
Guys, I am not sure whether I should be in this project, I think you added
the wrong person to this project.

*Narek Gevorgyan* | Technology / Product Guy

*Mobile: +1 512-661-1849*

*LinkedIn <https://www.linkedin.com/in/narekgev/> *

**Comment by Aveline-art** [comment]
@narek Thank you for letting me know. That was my mistake. I will make sure to be more careful in the future.

**Comment by narek** [comment]
sure np

*Narek Gevorgyan* | Technology / Product Guy

*Mobile: +1 512-661-1849*

*LinkedIn <https://www.linkedin.com/in/narekgev/> *

File: frontend/src/pages/landingPage/landingPage.js (hunk 3 above)
Comment (Aveline-art) [reply frontend/src/pages/landingPage/landingPage.js:6]: I've downloaded your branch and taken a look. Is this the only thing you did or did you also perform the other things I outlined above and fixed any resulting bugs?

If all else fails, I have made a PR here in your repo to compare the changes I have made to make the button responsive. Feel free to download the branch and let me know if you are still experiencing bugs. In that case, please show me gifs or pictures so I can see what might be going on.

Comment (NickTSA) [reply frontend/src/pages/landingPage/landingPage.js:6]: Screen Shot 2022-02-28 at 11 22 07 AM

I only changed the import method to "index.scss" that you mentioned. Everything else is the same.

Comment (NickTSA) [reply frontend/src/pages/landingPage/landingPage.js:6]: Okay, I think It worked now! Thank you

**Comment by NickTSA** [comment]
@Aveline-art finally! ready for review 😊

File: frontend/src/pages/landingPage/landingPage.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+          Together,
+          <br />
+          we can create greater civic change
+        </h1>
+        <p className="mb-5 paragraph-1 head-paragraph">
+          CivicTechJobs unites ambitious technology practitioners with volunteer
+          opportunities from a central hub of listings to build digital
+          products, programs, and services.{" "}
+        </p>
+        <Button size="lg" addClass="action-btn">
```
Comment (Aveline-art) [inline frontend/src/pages/landingPage/landingPage.js:30]: The button is missing a length prop to properly mirror Figma's design.

**Review by Aveline-art — CHANGES_REQUESTED** [review]
Great work @NickTSA ! I've tested things out a lot and I can see there was a lot of attention to detail.

I have listed a couple of corrections below. For the nitpicks, feel free to ignore them if it is too much work. They are easy enough to fix in the future, and I did not spot them until making a closer inspection with Figma. Some of the spacing too can also be slightly adjusted to fit Figma's design but those are also easy fixes.

One thing I want to point out is that usually images are absolute while texts and buttons overlays are relatively positioned. If you are up for it you can try it that way, if not already. It might improve readability and understanding of the code. It's not a big issue as far as I am concerned, though, since the code is relatively simple to understand

File: frontend/src/pages/landingPage/_landingPage.scss [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  }
+
+  position: relative;
+  background-image: url("./assets/images/svgs/group-background-pattern.svg");
+  background-repeat: no-repeat;
+  background-size: 100% 100%;
+  width: 100%;
+}
+
+.group-image {
```
Comment (Aveline-art) [inline frontend/src/pages/landingPage/_landingPage.scss:19]: One more thing: the class names are very non-specific. Since we have changed to loading all styles at once, it would cause a lot of trouble if we accidentally reuse a class name. Let's rename some of them to, for example, "landing-image" or "landing-mission-title", and so forth.

File: frontend/src/assets/images/images.js [hunk 6]
```diff
@@ -12,7 +12,7 @@ import copIconProduct from "./svgs/cop-icon-product.svg?url";
 import copIconUiux from "./svgs/cop-icon-uiux.svg?url";
 
 // Icons
-import IconDownArrow from "./svgs/icon-down-arrow.svg";
+import IconDownArrow from "./svgs/icon-down-arrow.svg?url";
```
[the remark this answers is not in the document]
Comment (NickTSA) [reply frontend/src/assets/images/images.js:15]: Okay I fixed this import. How do I add the class styles to the component? I tried "addClass", but it isn't working.

[the remark this answers is not in the document]
Comment (Aveline-art) [reply frontend/src/assets/images/images.js:15]: I'm not sure why this might be happening. Did you try taking a look at the Design System Helper?

File: frontend/src/pages/landingPage/_landingPage.scss [hunk 7]
```diff
@@ -0,0 +1,85 @@
+@import "../../components/Buttons/Buttons";
+@import "../../components/Basics/breakpoints";
+@import "../../components/Basics/typography";
```
[the remark this answers is not in the document]
Comment (NickTSA) [reply frontend/src/pages/landingPage/_landingPage.scss:3]: I've tried this previously. In this scenario it needs to be a percentage to scale properly with window resizing.

File: frontend/src/pages/landingPage/_landingPage.scss [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+.svg-group-layer {
+  margin: 0 12%;
+}
+
+.mission-section {
+  padding-top: 8%;
+  margin-top: -12%;
+  position: relative;
+  background-color: #ffefdb;
```
[the remark this answers is not in the document]
Comment (NickTSA) [reply frontend/src/pages/landingPage/_landingPage.scss:31]: Did this but classes aren't applying with "addClass". I will complete this push my branch so you can see it.

File: frontend/src/pages/landingPage/_landingPage.scss [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+.down-arrow {
+  width: 24px;
+  display: none;
+
+  @include media-max("smtablet") {
+    display: inline-block;
+  }
+
+  margin-bottom: 30px;
```
[the remark this answers is not in the document]
Comment (NickTSA) [reply frontend/src/pages/landingPage/_landingPage.scss:43]: not sure why this happen. We can ignore this file change.

**Comment by NickTSA** [comment]
Okay I fixed almost all the issues. The only thing left is the styles on the down-arrow. @Aveline-art

File: frontend/src/pages/landingPage/_landingPage.scss [hunk 10]
```diff
@@ -0,0 +1,85 @@
+@import "../../components/Buttons/Buttons";
+@import "../../components/Basics/breakpoints";
+@import "../../components/Basics/typography";
+
+.image-parent {
+  z-index: 1;
+  margin-top: -10%;
```
[the remark this answers is not in the document]
Comment (NickTSA) [reply frontend/src/pages/landingPage/_landingPage.scss:6]: I've tried this previously. In this scenario it needs to be a percentage to scale properly with window resizing.

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -29,7 +29,6 @@ We do not currently have a dedicated issue template for creating issues for cont
 5. Done! Now just wait for a team member to get back to you with feedback on your changes!
 
 > Important note 1: **We do not want to be flooded with pull requests!** Please, do **NOT** take on another issue or open a second pull request until your current pull request and linked issue are resolved by a team member.
-
```
[the remark this answers is not in the document]
Comment (NickTSA) [reply CONTRIBUTING.md:32]: not sure why this happen. We can ignore this file change.

File: frontend/src/assets/images/images.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (NickTSA) [reply frontend/src/assets/images/images.js]: I looked over the design system and followed the instructions to add a wrapper element. Now the classes are working except the "Width" is not applied to the down arrow component.

File: frontend/src/pages/landingPage/_landingPage.scss [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+.landing-mission-content {
+  margin: auto;
+  max-width: 800px;
+}
+
+.landing-top-content {
+  max-width: 705px;
+  margin: auto;
+  padding-left: 32px;
+  padding-right: 32px;
```
Comment (Aveline-art) [inline frontend/src/pages/landingPage/_landingPage.scss:52]: These paddings are in our design system

File: frontend/src/pages/landingPage/landingPage.js [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+          products, programs, and services.{" "}
+        </p>
+        <Button length="long" size="lg" addClass="landing-action-btn">
+          Join us
+        </Button>
+      </div>
+      <LandingImage />
+      <div className="landing-mission-section flex-container">
+        <div className="landing-mission-content px-4 align-center text-center">
+          <IconDownArrow addClass="down-arrow mb-4" />
```
Comment (Aveline-art) [inline frontend/src/pages/landingPage/landingPage.js:36]: The `addClass` prop does not do anything here since IconDownArrow is not defined in our codebase (it is actually imported through svgr, under the hood, which has a limited set of props, but addClass is not one of them).

I've hinted at the solution to this being in the our documentation, but you might have missed it. The optimal approach to do this would be to import the data-url version of the svg.

```jsx
<img src={iconDownArrow} width="23px" /> // See the difference between line 15 and 19 of images.js
```

Please message me via slack on what difficulties you had encountered in finding this solution so that we may improve the documentation.

File: CONTRIBUTING.md (hunk 11 above)
Comment (Aveline-art) [inline CONTRIBUTING.md:32]: Do these changes belong here? If not, please undo these changes.

**Review by Aveline-art — CHANGES_REQUESTED** [review]
Hi. Just a couple of small changes below.

I had assumed that some of the changes could be solved with a look at our documentation. But it seems that the documentation is not working for you? If so, please message me on Slack with suggestions and areas of improvements. It is vital that developers are able to work off of documentation so that I can give higher quality reviews and developers can make higher quality PRs.

**Comment by NickTSA** [comment]
@Aveline-art ready for review

**Review by Aveline-art — APPROVED** [review]
LGTM!
